src/api/app/controllers/concerns/webui/rescue_handler.rb

```diff
@@ -3,27 +3,18 @@ module Webui::RescueHandler
 
   included do
     rescue_from Pundit::NotAuthorizedError do |exception|
-      pundit_action = case exception.try(:query).to_s
-                      when 'index?' then 'list'
-                      when 'show?' then 'view'
-                      when 'create?' then 'create'
-                      when 'new?' then 'create'
-                      when 'update?' then 'update'
-                      when 'edit?' then 'edit'
-                      when 'destroy?' then 'delete'
-                      when 'create_branch?' then 'create_branch'
-                      else exception.try(:query)
-                      end
-      message = if pundit_action && exception.record
-                  "Sorry, you are not authorized to #{pundit_action} this #{exception.record.class}."
-                else
-                  'Sorry, you are not authorized to perform this action.'
-                end
+      message = unauthorized_message(exception)
+      redirect_path = unauthorized_path(exception)
+
       if request.xhr?
         render json: { error: message }, status: 400
```

Comment:
In case of anonymous `xhr` requests we return a `json` (instead of a `text`) response.
I see test passes so I'm supposing this is not a breaking change 😄 
Wdyt?